src/commands/UserCommand.ts

```diff
@@ -0,0 +1,74 @@
+import { Command } from "../Command";
+import { CommandInteraction, MessageEmbed } from "discord.js";
+import { SlashCommandBuilder } from "@discordjs/builders";
+
+const toUnix = (timestamp: number | Date) => {
+	if (timestamp instanceof Date) timestamp = timestamp.getTime();
+	return Math.floor(timestamp / 1000);
+};
+
+// set the class name, the export at the bottom and the file name to your desired command name (same as the one in the register function)
+class CommandName extends Command {
+	constructor() {
+		super("user"); // the name under which the bot internally stores your command (should be the same as the named set in `register`, must be unique)
+	}
+
+	async execute(interaction: CommandInteraction): Promise<void> {
+		// put the logic of your command here
+		// for example:
+		const user = interaction.options.getUser("user");
+		if (!user) return interaction.reply("Please provide a user.");
+		const member = interaction.guild?.members.cache.get(user.id);
+		const embed = new MessageEmbed()
+			.setTitle(`${user.tag} ${user.system ? "| system" : user.bot ? "| bot" : ""}`)
+			.setThumbnail(user.displayAvatarURL({ dynamic: true }))
+			.setDescription(`Ping: ${user.toString()}`);
+		if (user.flags?.toArray().length) {
+			embed.addField(
+				"Badges",
+				user.flags
+					.toArray()
+					.map((v) => v.toLowerCase().replace(/_/g, " "))
+					.join(", ")
+					.replace(/\b(.)/g, (c) => c.toUpperCase())
+			);
+		}
+		embed.addField("Id", user.id, true);
+		embed.addField("Created", `<t:${toUnix(user.createdTimestamp)}> (<t:${toUnix(user.createdTimestamp)}:R>)`);
+		embed.setTimestamp();
+		if (member) {
+			let roles = "‎";
+			member.roles.cache.forEach((role) => {
+				if (role.id === role.guild.id) return;
+				roles += `<@&${role.id}> `;
+			});
```

Comment:
This could probably be made nicer with map or reduce instead of forEach

Reply:
I probbaly could but won't